src/common/buffer.cc

```diff
@@ -1808,6 +1810,27 @@ void buffer::list::write_stream(std::ostream &out) const
   }
 }
 
+void buffer::list::compress(list& dest)
```

Comment:
I would suggest doing something like

 buffer::list::compress(int alg, list& dest);

and making a bunch of ALG_LZ4 etc constants in an enum so we can plug other algorithms in as well.